tests/core/test_model.py

```diff
@@ -245,17 +245,20 @@ def test_model_qualification():
 
 
 @pytest.mark.parametrize(
-    "partition_by_input, partition_by_output, expected_exception",
+    "partition_by_input, partition_by_output, output_dialect, expected_exception",
     [
-        ("a", ["`a`"], None),
-        ("(a, b)", ["`a`", "`b`"], None),
-        ("TIMESTAMP_TRUNC(`a`, DAY)", ["TIMESTAMP_TRUNC(`a`, DAY)"], None),
-        ("e", "", ConfigError),
-        ("(a, e)", "", ConfigError),
-        ("(a, a)", "", ConfigError),
+        ("a", ["`a`"], "bigquery", None),
+        ("(a, b)", ["`a`", "`b`"], "bigquery", None),
+        ("TIMESTAMP_TRUNC(`a`, DAY)", ["TIMESTAMP_TRUNC(`a`, DAY)"], "bigquery", None),
+        ("e", "", "bigquery", ConfigError),
+        ("(a, e)", "", "bigquery", ConfigError),
+        ("(a, a)", "", "bigquery", ConfigError),
+        ("(day(a),b)", ['DAY("a")', '"b"'], "trino", None),
```

Comment:
I added another parameter to this test to check that parsing was working as expected. Since this test was hardcoded to `dialect=bigquery`, I added another parameter for which dialect `partition_by_output` should be rendered in